File: app/src/main/java/com/tronography/pixelweather/model/CurrentWeatherBuilder.java
package com.tronography.pixelweather.model;

public class CurrentWeatherBuilder {
    private String city;
    private String country;
    private String description;
    private String icon;
    private int clouds;
    private int humidity;
    private double pressure;
    private double tempMax;
    private double tempMin;
    private double currentTemp;
    private double windSpeed;
    private long sunrise;
    private long sunset;
    private long dateTime;

    public CurrentWeatherBuilder setCity(String city) {
        this.city = city;
        return this;
    }

    public CurrentWeatherBuilder setCountry(String country) {
        this.country = country;
        return this;
    }

    public CurrentWeatherBuilder setDescription(String description) {
        this.description = description;
        return this;
    }

    public CurrentWeatherBuilder setIcon(String icon) {
        this.icon = icon;
        return this;
    }

    public CurrentWeatherBuilder setClouds(int clouds) {
        this.clouds = clouds;
        return this;
    }

    public CurrentWeatherBuilder setHumidity(int humidity) {
        this.humidity = humidity;
        return this;
    }

    public CurrentWeatherBuilder setPressure(double pressure) {
        this.pressure = pressure;
        return this;
    }

    public CurrentWeatherBuilder setTempMax(double tempMax) {
        this.tempMax = tempMax;
        return this;
    }

    public CurrentWeatherBuilder setTempMin(double tempMin) {
        this.tempMin = tempMin;
        return this;
    }

    public CurrentWeatherBuilder setCurrentTemp(double currentTemp) {
        this.currentTemp = currentTemp;
        return this;
    }

    public CurrentWeatherBuilder setWindSpeed(double windSpeed) {
        this.windSpeed = windSpeed;
        return this;
    }

    public CurrentWeatherBuilder setSunrise(long sunrise) {
        this.sunrise = sunrise;
        return this;
    }

    public CurrentWeatherBuilder setSunset(long sunset) {
        this.sunset = sunset;
        return this;
    }

    public CurrentWeatherBuilder setDateTime(long dateTime) {
        this.dateTime = dateTime;
        return this;
    }

    public CurrentWeatherModel build() {
        return new CurrentWeatherModel(city, country, description, icon, clouds, humidity, pressure, tempMax, tempMin, currentTemp, windSpeed, sunrise, sunset, dateTime);
    }
}

File: app/src/main/java/com/tronography/pixelweather/model/CurrentWeatherBuilder.kt
package com.tronography.pixelweather.model

class CurrentWeatherBuilder {
    private var city: String? = null
    private var country: String? = null
    private var description: String? = null
    private var icon: String? = null
    private var clouds: Int = 0
    private var humidity: Int = 0
    private var pressure: Double = 0.toDouble()
    private var tempMax: Double = 0.toDouble()
    private var tempMin: Double = 0.toDouble()
    private var currentTemp: Double = 0.toDouble()
    private var windSpeed: Double = 0.toDouble()
    private var sunrise: Long = 0
    private var sunset: Long = 0
    private var dateTime: Long = 0

    fun setCity(city: String): CurrentWeatherBuilder {
        this.city = city
        return this
    }

    fun setCountry(country: String): CurrentWeatherBuilder {
        this.country = country
        return this
    }

    fun setDescription(description: String): CurrentWeatherBuilder {
        this.description = description
        return this
    }

    fun setIcon(icon: String): CurrentWeatherBuilder {
        this.icon = icon
        return this
    }

    fun setClouds(clouds: Int): CurrentWeatherBuilder {
        this.clouds = clouds
        return this
    }

    fun setHumidity(humidity: Int): CurrentWeatherBuilder {
        this.humidity = humidity
        return this
    }

    fun setPressure(pressure: Double): CurrentWeatherBuilder {
        this.pressure = pressure
        return this
    }

    fun setTempMax(tempMax: Double): CurrentWeatherBuilder {
        this.tempMax = tempMax
        return this
    }

    fun setTempMin(tempMin: Double): CurrentWeatherBuilder {
        this.tempMin = tempMin
        return this
    }

    fun setCurrentTemp(currentTemp: Double): CurrentWeatherBuilder {
        this.currentTemp = currentTemp
        return this
    }

    fun setWindSpeed(windSpeed: Double): CurrentWeatherBuilder {
        this.windSpeed = windSpeed
        return this
    }

    fun setSunrise(sunrise: Long): CurrentWeatherBuilder {
        this.sunrise = sunrise
        return this
    }

    fun setSunset(sunset: Long): CurrentWeatherBuilder {
        this.sunset = sunset
        return this
    }

    fun setDateTime(dateTime: Long): CurrentWeatherBuilder {
        this.dateTime = dateTime
        return this
    }

    fun build(): CurrentWeatherModel {
        return CurrentWeatherModel(
                city!!,
                country!!,
                description!!,
                icon!!,
                clouds,
                humidity,
                pressure,
                tempMax,
                tempMin,
                currentTemp,
                windSpeed,
                sunrise,
                sunset,
                dateTime
        )
    }
}